pkgs/development/dotnet-modules/python-language-server/create_deps.sh

```diff
@@ -0,0 +1,29 @@
+
+# To use this script, clone the repo and set this variable to the checkout path:
+CHECKOUT_PATH=/home/tom/tools/microsoft-python-language-server/
+
+######################################################
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Generate lockfiles in source checkout
+cd $CHECKOUT_PATH/src
+dotnet nuget locals all --clear
+dotnet restore -v normal --no-cache PLS.sln --use-lock-file -r linux-x64
+
+# Use the lockfiles to make a file with two columns: name and version number
+# for all possible package dependencies
+cd $SCRIPTDIR
+echo "" > all_versions.txt
+for lockfile in $(find "$CHECKOUT_PATH" -name packages.lock.json); do
+    echo "Processing lockfile $lockfile"
+    python ./process_lockfile.py "$lockfile" >> all_versions.txt
+done
+# Add extra manually added packages
+cat ./manual_deps.txt >> all_versions.txt
+cat all_versions.txt | sed '/^$/d' | sort | uniq > tmp
+mv tmp all_versions.txt
+
+# Retrieve sha256 hashes for each dependency and format fetchNuGet calls into deps.nix
+./format-deps.sh all_versions.txt > deps.nix
```

Comment:
Is there a reason this is a separate script? Can it be inlined?

Reply:
The division into two scripts represents the two time-consuming parts of the process. Getting the dependencies exactly right can be tricky, which is why the `manual_deps.txt` file exists. For whatever reason, the script I wrote to scrape the lockfiles can still miss a few dependencies and you need to add these by hand. For doing this kind of thing, it's convenient to run `create_deps.sh`, then examine the `all_versions.txt` file it produces, then repeatedly tweak that and run `format_deps.sh`.

Reply:
Fair enough, makes sense to me. Although a better solution would be to find out why the scraper doesn't seem to pull in everything, but eh.

Reply:
Yeah that's kind of a mystery, the missing deps didn't seem to be anywhere when I grepped for them. I'm not a .Net expert so no idea why. Resolving this